src/libsync/owncloudpropagator.cpp

```diff
@@ -115,6 +124,61 @@ static bool blacklistCheck(SyncJournalDb* journal, const SyncFileItem& item)
     return newEntry.isValid();
 }
 
+void PropagateItemJob::itemDone(SyncFileItemPtr item, SyncFileItem::Status status, const QString &errorString)
```

Comment:
can some code be shared with PropagateItemJob::done() ? I guess you can simply call this function from done() .